File: app/schemas/booth.py
"""展位相关 Schema"""

from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class BoothCreate(BaseModel):
    """创建展位请求"""

    exhibition_id: int = Field(..., description="所属展会ID")
    booth_number: str = Field(..., min_length=1, max_length=50, description="展位编号")
    name: Optional[str] = Field(None, max_length=200, description="展位名称")
    description: Optional[str] = Field(None, description="展位描述")
    area: Optional[float] = Field(None, ge=0, description="展位面积（㎡）")
    price: Optional[float] = Field(None, ge=0, description="展位价格")
    floor: Optional[int] = Field(None, description="楼层")
    zone: Optional[str] = Field(None, max_length=50, description="展区")


class BoothUpdate(BaseModel):
    """更新展位请求"""

    name: Optional[str] = None
    description: Optional[str] = None
    area: Optional[float] = None
    price: Optional[float] = None
    status: Optional[str] = None
    exhibitor_id: Optional[int] = None


class BoothResponse(BaseModel):
    """展位响应"""

    id: int
    exhibition_id: int
    exhibition_name: Optional[str] = None
    exhibitor_id: Optional[int] = None
    exhibitor_name: Optional[str] = None
    booth_number: str
    name: Optional[str] = None
    description: Optional[str] = None
    area: Optional[float] = None
    price: Optional[float] = None
    floor: Optional[int] = None
    zone: Optional[str] = None
    status: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class BoothBookRequest(BaseModel):
    """预订展位请求"""

    booth_id: int = Field(..., description="展位ID")

File: expo-hub-frontend/src/api/booth.ts
import http from './index'
import type { ListResp } from './paged'
import { normList } from './paged'

export interface BoothItem {
  id: number
  exhibition_id: number
  exhibitor_id?: number
  booth_number: string
  name?: string
  description?: string
  area?: number
  price?: number
  floor?: number
  zone?: string
  size?: string
  location_area?: string
  company_name?: string
  exhibitor_name?: string
  status: string
  created_at: string
  updated_at: string
}

export interface CreateBoothParams {
  exhibition_id: number
  booth_number?: string
  name?: string
  description?: string
  area?: number
  price?: number
  floor?: number
  zone?: string
  size?: string
  location_area?: string
  company_name?: string
  status?: string
}

export interface BoothListParams {
  exhibition_id?: number
  zone?: string
  status?: string
  page?: number
  page_size?: number
}

export const boothApi = {
  async getList(params?: BoothListParams): Promise<ListResp<BoothItem>> {
    const res: any = await http.get('/booths', { params })
    return normList<BoothItem>(res)
  },
  async getMyBooths(params?: { page?: number; page_size?: number; status?: string }): Promise<ListResp<BoothItem>> {
    const res: any = await http.get('/booths/my', { params })
    return normList<BoothItem>(res)
  },
  getDetail(id: number): Promise<BoothItem> {
    return http.get(`/booths/${id}`) as Promise<BoothItem>
  },
  book(boothId: number): Promise<BoothItem> {
    return http.post('/booths/book', { booth_id: boothId }) as Promise<BoothItem>
  },
  apply(boothId: number): Promise<any> {
    return http.post(`/booths/${boothId}/apply`) as Promise<any>
  },
  // 主办方接口
  create(data: CreateBoothParams): Promise<BoothItem> {
    return http.post('/booths', data) as Promise<BoothItem>
  },
  update(id: number, data: Partial<CreateBoothParams>): Promise<BoothItem> {
    return http.put(`/booths/${id}`, data) as Promise<BoothItem>
  },
  assign(boothId: number, exhibitorId: number): Promise<BoothItem> {
    return http.post(`/booths/${boothId}/assign`, { exhibitor_id: exhibitorId }) as Promise<BoothItem>
  }
}

export type Booth = BoothItem
